src/client.ts

```diff
@@ -121,27 +121,85 @@ export class CognigyClient {
 	 * Convenience method to send a "resetState" event.
 	 */
 	public resetState(): Promise<string> {
-		if (!this.isConnected()) {
-			console.error("[Client] Error resetting state. You are not connected to the brain-server.");
-			return Promise.reject("[Client] Error resetting state. You are not connected to the brain-server.");
-		}
+		return new Promise((resolve, reject) => {
+			if (!this.isConnected()) {
+				console.error("[Client] Error resetting state. You are not connected to the brain-server.");
+				reject("[Client] Error resetting state. You are not connected to the brain-server.");
+			}
 
-		this.mySocket.emit("resetState", (currentState: string) => {
-			return Promise.resolve(currentState);
+			this.mySocket.emit("resetState", (currentState: string) => {
+				resolve(currentState);
+			});
 		});
+		
 	}
 
 	/**
 	 * Convenience method to send a "resetContext" event.
 	 */
 	public resetContext(): Promise<any> {
-		if (!this.isConnected()) {
-			console.error("[Client] Error resetting context. You are not connected to the brain-server.");
-			return Promise.reject("[Client] Error resetting contenxt. You are not connected to the brain-server.");
-		}
+		return new Promise((resolve, reject) => {
+			if (!this.isConnected()) {
+				console.error("[Client] Error resetting context. You are not connected to the brain-server.");
+				reject("[Client] Error resetting contenxt. You are not connected to the brain-server.");
+			}
+
+			this.mySocket.emit("resetContext", (currentContext: any) => {
+				resolve(currentContext);
+			});
+		});
+	}
+
+	/**
+	 * Convenience method to send a "injectContext" event.
+	 * 
+	 * @param context {any} Context object (should be valid JSON)
+	 */
+	public injectContext(context: any): Promise<string> {
+		return new Promise((resolve, reject) => {
+			if (!this.isConnected()) {
+				console.error("[Client] Error injecting context. You are not connected to the brain-server.");
+				reject("[Client] Error injecting context. You are not connected to the brain-server.");
+			}
+
 
-		this.mySocket.emit("resetContext", (currentContext: any) => {
-			return Promise.resolve(currentContext);
+			if (typeof context !== 'object') {
+				// passed context is not a JSON object (or any object), trying to convert
+				try {
+					context = JSON.parse(context);
+				} catch (err) {
+					console.error("[Client] Error injecting context. The passed context is not JSON.");
+					reject("[Client] Error injecting context. The passed context is not JSON.");
+				}
+			}
+
+			this.mySocket.emit("injectContext", context, (currentState: string) => {
```

Comment:
Some as within "CognigyClient"-repo. We should change the name "currentState" to something like "currentContext" and use the correct type (since I dont thinkg that we are using string for context)